cf/data/dask_regrid.py

```diff
@@ -562,12 +567,45 @@ def _regrid(
 
             del indptr, pos_data
 
+        elif method == "nearest_dtos":
+            # 3) Nearest neighbour dtos method:
+            #
+            # Set to 0 any weight that corresponds to a masked source
```

Comment:
To check my understanding of this via the maths, am I correct in deducing that the ultimate dot product deals with the fact that, if we set to zero for masked source points (which is the only sensible option IMO so I agree is the way) then we are effectively using less points as sources, i.e. instead of shape (J, I) weights matrix as described here:

>         weights: `scipy.sparse.spmatrix`
>            The sparse weights matrix that defines the regridding
>            operation. Might be modified (not in-place) to account for
>            missing data in *a*. Must have shape ``(J, I)``, where
>            ``J`` is the number of destination grid cells and ``I`` is
>            the number of source grid cells. 

we now really have (J, I - M) where M is the number of masked source points? So I can convince myself the reduction in total source points is accounted for in the calculation. Reminding myself about dot products I think it will, but it would be good to get confirmation. Thanks.

Reply:
Hi Sadie. Missing-data source points are effectively ignored by setting their corresponding weights to zero. Those zero-valued weights still need to appear in the `(J, I)` shaped matrix of weights, otherwise the dot product wont work, because the missing elements still exist in the source data (which by this stage has shape `(I,)`.

Reply:
OK I see, thanks for clarifying. All good!